server/resourcemanager/encoding.go

```diff
@@ -0,0 +1,68 @@
+package resourcemanager
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"gopkg.in/yaml.v2"
+)
+
+type encoder interface {
+	Marshal(in interface{}) (out []byte, err error)
+	Unmarshal(in []byte, out interface{}) (err error)
+	Accepts(contentType string) bool
+	ResponseContentType() string
+}
+
+var encoders = []encoder{
+	jsonEncoder,
+	yamlEncoder,
+}
+
+var jsonEncoder = basicEncoder{
+	contentType: "application/json",
+	marshalFn:   json.Marshal,
+	unmarshalFn: json.Unmarshal,
+}
+
+var yamlEncoder = basicEncoder{
+	contentType: "text/yaml",
+	marshalFn:   yaml.Marshal,
+	unmarshalFn: yaml.Unmarshal,
+}
+
+type basicEncoder struct {
+	contentType string
+	marshalFn   func(interface{}) ([]byte, error)
+	unmarshalFn func([]byte, interface{}) error
+}
+
+func (e basicEncoder) ResponseContentType() string {
+	return e.contentType
+}
+
+func (e basicEncoder) Accepts(contentType string) bool {
+	return contentType == e.contentType
+}
+
+func (e basicEncoder) Marshal(in interface{}) (out []byte, err error) {
+	return e.marshalFn(in)
+}
+func (e basicEncoder) Unmarshal(in []byte, out interface{}) (err error) {
+	return e.unmarshalFn(in, out)
+}
+
+var errUnacceptableContentType = errors.New("unacceptable content type")
+
+func encoderFromRequest(r *http.Request) (encoder, error) {
```

Comment:
this func is used to get the correct encoder instance for a given request based on the content type